### PR TITLE
Refactor sign up page tests

### PR DESCRIPTION
**Pull request by VoK0**
* implement short imports 
* rewrite signUp tests 
* implement Mailosaur 
* create separete tsconfig.json -- just for UI-tests

### CODE REVIEW

File: ui-tests/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "baseUrl": ".",
+    "paths": {
+      "@pages/*": ["pageObjects/*"],
+      "@testtypes/*": ["support/types/*"],
```
Comment (peterSirotnak) [inline ui-tests/tsconfig.json:8]: please use camelCase @testTypes

File: ui-tests/tests/auth/signUp.spec.ts [hunk 2]
```diff
@@ -19,7 +29,7 @@ test.describe('Spec file for Sign Up tests for customers', async () => {
 
     await page.goto('');
     // Verify URL
-    await expect(page).toHaveURL(`${baseURL + signUpPage.links.loginAddressLink}`);
+    await expect(page).toHaveURL(`${`${baseURL}/${signUpPage.links.loginAddressLink}`}`);
```
Comment (peterSirotnak) [inline ui-tests/tests/auth/signUp.spec.ts:32]: This url is not correct

File: ui-tests/support/types/signUpFormErros.interface.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export interface SignUpFormErrors {
+    email?: boolean;
+    password?: boolean;
+    firstName?: boolean;
+    lastName?: boolean;
+  }
```
Comment (peterSirotnak) [inline ui-tests/support/types/signUpFormErros.interface.ts:6]: new line at the end of the file

**Review by peterSirotnak — CHANGES_REQUESTED** [review]
Please fix comments

File: ui-tests/tests/auth/signUp.spec.ts [hunk 4]
```diff
@@ -31,4 +41,76 @@ test.describe('Spec file for Sign Up tests for customers', async () => {
...
+    // Prepare data 
+    const signUpPage = new SignUpPage(page);
+    succUser.email = signUpPage.getMailosaurEmailAddress(succUser);
+    // Fullfill the sign up form and register
+    await page.goto('');
+    await signUpPage.locators.createOneLink.click();
+    await signUpPage.fillOutSignUpUserDetails(succUser);
+    await signUpPage.locators.registerButton.click();
+    await expect(signUpPage.locators.verificationEmailSentTitleLoc).toHaveText(
+      signUpPage.messages.verificationEmailSentTitle,
```
Comment (peterSirotnak) [inline ui-tests/tests/auth/signUp.spec.ts:99]: Test fails:
Error: expect(received).toHaveText(expected)

    Expected string: "Verification email sent"
    Received string: ""
    Call log:
      - expect.toHaveText with timeout 5000ms
      - waiting for selector "h2.title"

File: ui-tests/tests/auth/signUp.spec.ts [hunk 5]
```diff
@@ -31,4 +41,76 @@ test.describe('Spec file for Sign Up tests for customers', async () => {
...
+    await signUpPage.locators.inputLastName.fill('');
+    await signUpPage.locators.marketingCheckbox.check();
+    await signUpPage.locators.tosCheckbox.check();
+    await signUpPage.locators.registerButton.click();
+    // Verify error message toast message.
+    await expect(signUpPage.locators.registrationAlertMessage).toHaveText(
+      signUpPage.messages.validationErrorAlert,
+    );
+    // Verify number of errors from input fields and verify the error message.
+    const inputErrorMessagesTexts = await signUpPage.locators.fieldFormValidatior.evaluateAll(
```
Comment (peterSirotnak) [inline ui-tests/tests/auth/signUp.spec.ts:78]: fieldFormValidatior -> fieldFormValidator just a typo

File: ui-tests/tsconfig.json (hunk 1 above)
Comment (VoK0) [reply ui-tests/tsconfig.json]: Fixed

File: ui-tests/tests/auth/signUp.spec.ts [hunk 6]
```diff
@@ -31,4 +41,76 @@ test.describe('Spec file for Sign Up tests for customers', async () => {
...
+    // Verifyt that mailosaur have that email in the box.
+    const mailosaur = new MailosaurClient(`${process.env.MAILOSAUR_API_KEY}`);
+    const serverId = process.env.MAILOSAUR_UI_TESTS_SERVER_ID;
+    const searchCriteria = {
+      sentTo: `${succUser.email}`};
+    const message = await mailosaur.messages.get(serverId, searchCriteria);
+    // Click on the link
+    const importantLink = message.html.links[0].href;
+    // visit an emailed link
+    page.goto(importantLink);
```
Comment (peterSirotnak) [inline ui-tests/tests/auth/signUp.spec.ts:113]: please add await for page.goto(importantLink); and verification that user is on correct page.

File: ui-tests/tests/auth/signUp.spec.ts (hunk 5 above)
Comment (VoK0) [reply ui-tests/tests/auth/signUp.spec.ts]: Fixed

File: ui-tests/tests/auth/signUp.spec.ts (hunk 4 above)
Comment (VoK0) [reply ui-tests/tests/auth/signUp.spec.ts:102]: I am not able to replicate this issue :/

File: ui-tests/tests/auth/signUp.spec.ts (hunk 6 above)
Comment (VoK0) [reply ui-tests/tests/auth/signUp.spec.ts]: Fixed.

File: ui-tests/tests/auth/signUp.spec.ts [hunk 7]
```diff
@@ -1,14 +1,24 @@
...
+import MailosaurClient from 'mailosaur';
 import { getUser } from '../../../cypress/pages/auth/getUser';
-import { SignUpPage } from '../../pageObjects/signUp.page';
-import User from '../../support/types/user.interface';
 
 test.describe('Spec file for Sign Up tests for customers', async () => {
+  const apiCalls = new ApiCalls();
   let adminUser: User;
+  let casualUser: User;
+  let succUser: User;
```
Comment (peterSirotnak) [inline ui-tests/tests/auth/signUp.spec.ts:14]: plese delete this user after tests

**Review by peterSirotnak — APPROVED** [review]
LGTM

**Pull request merged**